src/repositories/participants.ts

```diff
@@ -0,0 +1,80 @@
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Mojaloop Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Mojaloop Foundation
+ - Name Surname <name.surname@mojaloop.io>
+
+ * Google
+ - Steven Wijaya <stevenwjy@google.com>
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+ --------------
+ ******/
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import firebase from '~/lib/firebase'
+import { logger } from '~/shared/logger'
+
+export interface IParticipantRepository {
+  /**
+   * Replace existing participants list with new list.
+   *
+   * @param data   Documents that are about to be added.
+   */
+  replace(data: Record<string, any>): Promise<void>
+}
+
+export class FirebaseParticipantRepository implements IParticipantRepository {
+  // TODO: Confirm data Type
+  async replace(data: Record<string, any>): Promise<void> {
+    const collectionRef: FirebaseFirestore.CollectionReference<FirebaseFirestore.DocumentData> = firebase
+      .firestore()
+      .collection('participants')
+
+    // Find and update all matching documents in Firebase that match the given conditions.
+    await collectionRef
+      .get()
+      .then(async (response) => {
+        // Create a batch to perform all of the updates using a single request.
+        // Firebase will also execute the updates atomically according to the
+        // API specification.
+        const batch = firebase.firestore().batch()
+
+        const batchSize = response.size
+        if (batchSize > 0) {
+          // If previous participants list exists, delete it
+
+          // Iterate through all matching documents add them to the processing batch.
+          response.docs.forEach((doc) => {
+            batch.delete(doc.ref)
+          })
+        }
+        // Iterate through received participants list and add them to the processing batch.
+        data.array.forEach((participant: any) => {
+          batch.set(collectionRef.doc(), participant)
```

Comment:
I think it would be better to use `batch.add` rather than `batch.set` here

Reply:
batch.add doesn't exist unfortunately

Reply:
Ohhh I see, didn't know about that... okay then :-)